engine/hatchery/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,331 @@
+		{Name: "CDS_TOKEN", Value: h.Configuration().API.Token},
+		{Name: "CDS_SINGLE_USE", Value: "1"},
+		{Name: "CDS_MODEL", Value: fmt.Sprintf("%d", spawnArgs.Model.ID)},
+		{Name: "CDS_HATCHERY", Value: fmt.Sprintf("%d", h.hatch.ID)},
+		{Name: "CDS_HATCHERY_NAME", Value: h.hatch.Name},
+		{Name: "CDS_FORCE_EXIT", Value: "1"},
+		{Name: "CDS_TTL", Value: fmt.Sprintf("%d", h.Config.WorkerTTL)},
+	}
+
+	var logJob string
+	if spawnArgs.JobID > 0 {
+		if spawnArgs.IsWorkflowJob {
+			logJob = fmt.Sprintf("for workflow job %d,", spawnArgs.JobID)
+			envs = append(envs, apiv1.EnvVar{Name: "CDS_BOOKED_WORKFLOW_JOB_ID", Value: fmt.Sprintf("%d", spawnArgs.JobID)})
+		} else {
+			logJob = fmt.Sprintf("for pipeline build job %d,", spawnArgs.JobID)
+			envs = append(envs, apiv1.EnvVar{Name: "CDS_BOOKED_PB_JOB_ID", Value: fmt.Sprintf("%d", spawnArgs.JobID)})
+		}
+	}
+
+	memory := int64(h.Config.DefaultMemory)
+	for _, r := range spawnArgs.Requirements {
+		if r.Type == sdk.MemoryRequirement {
+			var err error
+			memory, err = strconv.ParseInt(r.Value, 10, 64)
+			if err != nil {
+				log.Warning("spawnKubernetesDockerWorker> %s unable to parse memory requirement %s:%s", logJob, memory, err)
+				return "", err
+			}
+		}
+	}
+
+	var gracePeriodSecs int64
+	pod, err := h.k8sClient.CoreV1().Pods(h.Config.KubernetesNamespace).Create(&apiv1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			DeletionGracePeriodSeconds: &gracePeriodSecs,
+			Labels: map[string]string{
+				LABEL_WORKER:        label,
+				LABEL_WORKER_MODEL:  strings.ToLower(spawnArgs.Model.Name),
+				LABEL_HATCHERY_NAME: h.Configuration().Name,
+			},
+		},
+		Spec: apiv1.PodSpec{
+			RestartPolicy:                 apiv1.RestartPolicyNever,
+			TerminationGracePeriodSeconds: &gracePeriodSecs,
+			Containers: []apiv1.Container{
+				{
+					Name:  name,
+					Image: spawnArgs.Model.Image,
+					Env:   envs,
+					Resources: apiv1.ResourceRequirements{
+						Requests: apiv1.ResourceList{
+							apiv1.ResourceMemory: resource.MustParse(fmt.Sprintf("%d", memory)),
+						},
+					},
+				},
+			},
+		},
+	})
+
+	return pod.Name, err
+}
+
+// WorkersStarted returns the number of instances started but
+// not necessarily register on CDS yet
+func (h *HatcheryKubernetes) WorkersStarted() int {
+	workersLen := 0
+	list, err := h.k8sClient.CoreV1().Pods(h.Config.KubernetesNamespace).List(metav1.ListOptions{LabelSelector: LABEL_HATCHERY_NAME})
+	if err != nil {
+		return workersLen
+	}
+	for _, pod := range list.Items {
+		labels := pod.GetLabels()
+		if labels[LABEL_HATCHERY_NAME] == h.Configuration().Name {
+			workersLen++
+		}
+	}
+
+	return workersLen
+}
+
+// WorkersStartedByModel returns the number of instances of given model started but
+// not necessarily register on CDS yet
+func (h *HatcheryKubernetes) WorkersStartedByModel(model *sdk.Model) int {
+	list, err := h.k8sClient.CoreV1().Pods(h.Config.KubernetesNamespace).List(metav1.ListOptions{LabelSelector: LABEL_WORKER_MODEL})
```

Comment:
this func returns "WorkersStartedByModel returns the number of instances of given model", but model arg is never used.